### PR TITLE
Modernize addon-dummy-generate-test.js

### PR DESCRIPTION
**Pull request by RichardOtvos**
Epic: #8511

### CODE REVIEW

File: tests/acceptance/addon-dummy-generate-test.js [hunk 1]
```diff
@@ -49,218 +47,199 @@ describe('Acceptance: ember generate in-addon-dummy', function() {
...
 
-  function generateInAddon(args) {
+  async function generateInAddon(args) {
     let generateArgs = ['generate'].concat(args);
 
-    return initAddon().then(function() {
-      return ember(generateArgs);
-    });
+    await initAddon();
+    return ember(generateArgs);
```
Comment (Turbo87) [inline tests/acceptance/addon-dummy-generate-test.js]:
```suggestion
    await ember(generateArgs);
```

File: tests/acceptance/addon-dummy-generate-test.js [hunk 2]
```diff
@@ -1,11 +1,10 @@
 'use strict';
 
-const co = require('co');
-const RSVP = require('rsvp');
 const ember = require('../helpers/ember');
 const fs = require('fs-extra');
+const util = require('util');
 const path = require('path');
-let remove = RSVP.denodeify(fs.remove);
+let remove = util.promisify(fs.remove);
```
Comment (Turbo87) [inline tests/acceptance/addon-dummy-generate-test.js]: `fs-extra` supports Promise-style invocation by default now, so this wrapper is not needed anymore.

File: tests/acceptance/addon-dummy-generate-test.js [hunk 3]
```diff
@@ -27,20 +24,19 @@ describe('Acceptance: ember generate in-addon-dummy', function() {
...
-  );
+  beforeEach(async function() {
+    let tmpdir = await mkTmpDirIn(tmproot);
+    process.chdir(tmpdir);
+  });
 
   afterEach(function() {
     process.chdir(root);
-    return remove(tmproot);
+    return fs.remove(tmproot);
```
Comment (Turbo87) [inline tests/acceptance/addon-dummy-generate-test.js]:
```suggestion
    await fs.remove(tmproot);
```

File: tests/acceptance/addon-dummy-generate-test.js [hunk 4]
```diff
@@ -27,20 +24,19 @@ describe('Acceptance: ember generate in-addon-dummy', function() {
...
-      let tmpdir = yield mkTmpDirIn(tmproot);
-      process.chdir(tmpdir);
-    })
-  );
+  beforeEach(async function() {
+    let tmpdir = await mkTmpDirIn(tmproot);
+    process.chdir(tmpdir);
+  });
 
   afterEach(function() {
```
Comment (Turbo87) [inline tests/acceptance/addon-dummy-generate-test.js]:
```suggestion
  afterEach(async function() {
```

**Comment by locks** [comment]
Thanks for the help! I'll be closing this PR as most of the changes seem to already have gotten into the codebase. I rebased one of your other PRs and it's just waiting for CI to pass :)

**Pull request closed without merging**